docs/src/design/stack/io_ops.md

```diff
@@ -0,0 +1,193 @@
+
+The effect of this operation on the rest of the stack is:
+* **Right shift** starting from position $0$.
+
+### SDEPTH
+Assume $a$ is the current depth of the stack stored in the stack bookkeeping register $b_0$ (as described [here](./main.md#stack-representation)). The `SDEPTH` pushes $a$ onto the stack. The diagram below illustrates this graphically.
+
+![sdepth](../../assets/design/stack/io_ops/SDEPTH.png)
+
+Stack transition for this operation must satisfy the following constraints:
+
+>$$
+s_0' - b_0 = 0 \text{ | degree} = 1
+$$
+
+The effect of this operation on the rest of the stack is:
+* **Right shift** starting from position $0$.
+
+### READ
+Assume $a$ is an element at the head of the advice tape. The `READ` operation removes $a$ from the advice tape and pushes it onto the stack. The diagram below illustrates this graphically.
+
+![read](../../assets/design/stack/io_ops/READ.png)
+
+The `READ` operation does not impose any constraints against the first element of the stack.
+
+The effect of this operation on the rest of the stack is:
+* **Right shift** starting from position $0$.
+
+### READW
+Assume $a$, $b$, $c$, and $d$, are the elements at the head of the advice tape (with $a$ being on top). The `READW` operation removes these elements from the advice tape and puts them onto the stack by overwriting the top $4$ stack elements. The diagram below illustrates this graphically.
+
+![readw](../../assets/design/stack/io_ops/READW.png)
+
+The `READW` operation does not impose any constraints against the top $4$ elements of the stack.
+
+The effect of this operation on the rest of the stack is:
+* **No change** starting from position $4$.
+
+## Memory access operations
+Miden VM exposes several operations for reading from and writing to random access memory. Memory in Miden VM is managed by the [Memory chiplet](../chiplets/memory.md).
+
+Communication between the stack and the memory chiplet is accomplished via the chiplet bus $b_{chip}$. To make requests to the chiplet bus we need to divide its current value by the value representing memory access request. The structure of memory access request value is described [here](../chiplets/memory.md#memory-row-value).
+
+To enforce the correctness of memory access, we can use the following constraint:
+
+>$$
+b_{chip}' \cdot u_{mem} = b_{chip} \text{ | degree } = 2
+$$
+
+In the above, $u_{mem}$ is the value of memory access request. Thus, to describe AIR constraint for memory operations, it is sufficient to describe how $u_{mem}$ is computed. We do this in the following sections.
+
+### MLOADW
+Assume that the words with elements $v_0, v_1, v_2, v_3$ is located in memory at address $a$. The `MLOADW` operation pops an element off the stack, interprets it as a memory address, and replaces the remaining 4 elements at the top of the stack with values located at the specified address. The diagram below illustrates this graphically.
```

Comment:
nit: "Assume that the word with elements..." (This comes up again below at least once)

Reply:
Fixed!